server/build-plugins.js

```diff
@@ -33,6 +35,47 @@ const getPackageName = name => {
   }
 };
 
+async function installRemotePackages(installPackages) {
+  // check if connected to internet
+  // if not, skip npm install
+  const EXTERNAL_URI = process.env.EXTERNAL_URI || 'npmjs.com';
+  await require('dns').lookup(EXTERNAL_URI, async err => {
+    if (err && err.code === 'ENOTFOUND')
+      logger.error(`Can't reach npm servers. Skipping npm install`);
+    else {
+      logger.info('Installing plugin packages...');
+      await execSync(
+        `npm install --no-save ${installPackages.join(' ')} --production`,
```

Comment:
why the `--production` flag?

Reply:
I don't remember to be honest. `npm install`ing via `execSync` was giving me some trouble when I was originally implementing and I remember this being in a suggestion I found online. My guess is it doesn't totally matter but conceptually I can see it being good to have the production version of a plugin we're getting remotely and bundling in with our app.

Reply:
build-plugins is never really ever run by the user but rather by webpack itself. It's possible to pass options through there but it would be through the same process as BPANEL_PREFIX which is pretty convoluted but at least is related to the runtime you are starting when you would pass it. My feeling is that this is pretty far away from the user and not something they should be having to worry about, remote plugins basically being assumed production and if you want otherwise, the local_plugins gives you much more flexibility.